eng/build-job.yml

```diff
@@ -13,7 +13,7 @@ parameters:
 
 ### Product build
 jobs:
-- template: xplat-job.yml
+- template: xplat-test-job.yml
```

Comment:
The purpose of including xplat-test-job in build-job is to avoid duplicating the construction of productBuildArtifactName / productBuildRootFolderPath. When I was starting to work on CI infra, I originally tried to put some such variables into xplat-job but I was told that xplat-job is considered too general so I moved them into xplat-test-job. Have you got any suggestions as to a better name? xplat-pipeline-job, for instance?

Reply:
I like `xplat-pipeline-job.yml`.